query/src/main/scala/filodb/query/exec/MetadataExecPlan.scala

```diff
@@ -547,15 +560,28 @@ final case class TsCardExec(queryContext: QueryContext,
 
     source.checkReadyForQuery(dataset, shard, querySession)
     source.acquireSharedLock(dataset, shard, querySession)
-
+    val clusterNameLowercase = clusterName.toLowerCase()
     val rvs = source match {
       case tsMemStore: TimeSeriesStore =>
         Observable.eval {
           val cards = tsMemStore.scanTsCardinalities(
             dataset, Seq(shard), shardKeyPrefix, numGroupByFields)
           val it = cards.map { card =>
-            CardRowReader(prefixToGroup(card.prefix),
-              CardCounts(card.value.activeTsCount, card.value.tsCount))
+            val groupKey = prefixToGroupWithClusterAndDataset(card.prefix, clusterName, dataset.dataset)
+
+            // NOTE: cardinality data from downsample cluster is stored as total count in CardinalityStore. But for the
+            // user perspective, the cardinality data in downsample is a longterm data. Hence, we are forking the
+            // data path based on the cluster the data is being served from
+            if (clusterNameLowercase.contains("downsample")) {
```

Comment:
Ask if this is the best way check for downsample cluster data ?

Reply:
For posterity: we are using the cluster name because the alternative would be to use the type of `source`, which is a bit more dangerous.